contracts/Governance/VestingToken.sol

```diff
@@ -0,0 +1,119 @@
+    address public beneficiary; // address of the beneficiary
+    uint256 public withdrawn; // quantity of token withdrawn so far
+
+    modifier notInterrupted() {
+        require(!interrupted, "The contract has been interrupted");
+        _;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owner can do this");
+        _;
+    }
+
+    modifier onlyBeneficiary() {
+        require(msg.sender == beneficiary, "Only beneficiary can do this");
+        _;
+    }
+
+    modifier vestingStarted() {
+        require(isVestingStarted, "Vesting must be started");
+        _;
+    }
+
+    /// @param _token Address of token being vested
+    constructor(address _token, address _owner) {
+        token = ERC20(_token);
+        owner = _owner;
+    }
+
+    /// @notice Returns the token amount that is currently withdrawable
+    /// @return withdrawable Quantity of withdrawable Radicle asset
+    function withdrawableBalance() public view returns (uint256 withdrawable) {
+        uint256 timePassed = block.timestamp.sub(vestingStartTime);
+
+        if (timePassed < vestingPeriod) {
+            uint256 vested = totalVestingAmount.mul(timePassed) / vestingPeriod;
+            withdrawable = vested.sub(withdrawn);
+        } else {
+            withdrawable = totalVestingAmount.sub(withdrawn);
+        }
+    }
+
+    /// @notice Grant vesting tokens to the beneficiary.
+    ///
+    /// @param ofBeneficiary Address of beneficiary
+    /// @param ofTokenAmount Amount of tokens
+    /// @param ofVestingPeriod Vesting period in seconds from vestingStartTime
+    function grantTokens(
+        address ofBeneficiary,
+        uint256 ofTokenAmount,
+        uint256 ofVestingPeriod
+    ) external onlyOwner notInterrupted {
+        require(!isVestingStarted, "VestingToken::grantTokens: vesting already started");
+        require(ofTokenAmount > 0, "VestingToken::grantTokens: amount must be positive");
+        require(
+            token.transferFrom(msg.sender, address(this), ofTokenAmount),
+            "VestingToken::grantTokens: token deposit failed"
+        );
+
+        isVestingStarted = true;
+        vestingStartTime = block.timestamp;
+        totalVestingAmount = ofTokenAmount;
+        vestingPeriod = ofVestingPeriod;
+        beneficiary = ofBeneficiary;
+    }
+
+    /// @notice Withdraw vested tokens
+    function withdrawVested() external onlyBeneficiary vestingStarted notInterrupted {
+        uint256 withdrawable = withdrawableBalance();
+
+        withdrawn = withdrawn.add(withdrawable);
+
+        require(
+            token.transfer(beneficiary, withdrawable),
+            "VestingToken::withdrawVested: transfer to beneficiary failed"
+        );
+    }
+
+    /// @notice Force withdrawal of vested tokens to beneficiary
+    /// @notice Send remainder back to owner
+    /// @notice Prevent further vesting
+    function terminateVesting() external onlyOwner vestingStarted notInterrupted {
```

Comment:
The current API won't be very useful for our use case. This function should accept the timestamp of when the termination should be enacted.

Reply:
This terminates it immediately, for example in the case an employee leaves the company. We can call it on the date they leave, or do you think that's a problem?

Reply:
We know in advance when exactly a person is going to leave the company. The current API doesn't help us with that use case, it requires somebody to actually send a message roughly at that moment. It's imprecise and cumbersome. The smart contract could do that for us automatically and with perfect precision.

Reply:
Will tackle this in a separate PR